01_host/03_transition/01_runtime-interaction.adoc

```diff
@@ -130,54 +133,43 @@ of 64KB. This memory should be made available to the Polkadot Runtime
+(func $runtime_entrypoint (param $data i32) (param $len i32) (result i64))
 ----
 
-In each invocation of a Runtime entry, the argument(s) which are supposed to be
-sent to the entry, need to be SCALE encoded into a byte array stem:[B]
+In each invocation of a Runtime entrypoints, the argument(s) which are supposed to be
+sent to the entrypoint, need to be SCALE encoded into a byte array stem:[B]
 (<<sect-scale-codec>>) and copied into a section of Wasm shared memory managed
 by the shared allocator described in <<sect-memory-management>>.
 
-When the Wasm method , corresponding to the entry, is invoked, two
+When the Wasm method, corresponding to the entrypoint, is invoked, two
 integers are passed as arguments. The first argument is set to the
 memory address of the byte array stem:[B] in Wasm memory. The
 second argument sets the length of the encoded data stored in
 stem:[B].
 
 [#sect-runtime-return-value]
-===== Receiving Data from a Runtime Entry
+===== Receiving Data from a Runtime Entrypoint
 
 The value which is returned from the invocation is an integer,
 representing two consecutive integers in which the least significant one
-indicates the pointer to the offset of the result returned by the entry
+indicates the pointer to the offset of the result returned by the entrypoint
 encoded in SCALE codec in the memory buffer. The most significant one
 provides the size of the blob.
 
-[#sect-handling-runtime-state-update]
-===== Handling Runtimes update to the State
-
-In order for the Runtime to carry on various tasks, it manipulates the current
-state by means of executing calls to various Polkadot Host APIs
-(<<chap-host-api>>). It is the duty of Host APIs to determine the context in
-which these changes should persist. For example, if Polkdot Host needs to
-validate a transaction using entry (<<sect-rte-validate-transaction>>), it needs
-to sandbox the changes to the state just for that Runtime call and prevent the
-global state of the system from being influence by the call to such a Runtime
-entry. This includes reverting the state of function calls which return errors
-or panic.
-
-As a rule of thumb, any state changes resulting from Runtime entries are not
-persistent with the exception of state changes resulting from calling
-`Core_execute_block` (<<sect-rte-core-execute-block>>) while Polkadot Host is
-importing a block (<<sect-block-validation>>).
-
-For more information on managing multiple variant of state see
-<<sect-managing-multiple-states>>.
+[#sect-runtime-version-custom-section]
+===== Runtime Version Custom Section
+
+For newer Runtimes, the Runtime version (<<defn-rt-core-version>>) can be read
+directly from the Wasm custom section with the name `runtime_version`. The
+content is a SCALE encoded structure as described in <<defn-rt-core-version>>.
+
+Retrieving the Runtime version this way is preferred over calling the
+`Core_version` entrypoint since it involves significantly less overhead.
```

Comment:
I think it would make mores sense to add this to the section for runtime constant.

Reply:
I'm not sure I agree, it's not exactly a constant exposed by the Runtime/Wasm but the entry of the Wasm custom section. I think having it in the "Interacting with the Runtime" section is appropriate.  (I updated this PR with a link to the Wasm custom section spec, just FYI)

Reply:
Ok, then I missunderstood. Perfect, thanks.